### ISSUE TITLE
Roadmap For 1.4.0

### ISSUE THREAD

**Issue by Hypocrita20XX** [Hypocrita20XX/Conexus#25]
The more I think about it, the less I see any reason why the user needs to worry about mode selection.
Ideally Conexus should be able to handle this on its own without any need for input.

Will look into this for the proper 1.3.1 release.

**Comment by Hypocrita20XX** [comment]
> Merge modes into one, under-the-hood function
> Modify UI to accommodate mode merge
> Modify list of links log message to check for existence of file in Documents\Conexus\Logs, if not, let user know to check if Links.txt exists in that location
> Go through code and add any remaining missing exception catching

> Finally update the video tutorials, I'm very much behind on this due to some medication changes that put me out of action for several weeks, hoping I'll be back on track this week

**Comment by Hypocrita20XX** [comment]
Also, figure out what to do with the video tutorial archives. Probably Nexus Mods. They're giving me, and others, issues, so they need to shove off for a while.

**Comment by Hypocrita20XX** [comment]
This is probably going to be 1.3.2, now that I think about it, this goes beyond simple bug fixes.

Also, more importantly, I'm going to put together some basic dev documentation, now that there's a fork, this seems like a good idea. For now, just stuff related to the UI and how to modify it, because that's immediately useful to me, but other stuff might/will follow as I think of it.

**Comment by Hypocrita20XX** [comment]
Wait that would be 1.4. Eh, thinking.
In any case, modes have been merged seemingly successfully.

**Comment by Hypocrita20XX** [comment]
Alright, functionally, everything is fine.
Exception checks are all that's left, for Links file, verify that there is a username/password, that SteamCMD actually downloaded mods.

**Comment by Hypocrita20XX** [comment]
Note so I don't forget: quickstart will need updated

**Comment by Hypocrita20XX** [comment]
1.4.0 is almost release-ready.
Needs a bit more testing, then hopefully that'll be it.

**Comment by Hypocrita20XX** [comment]
I wouldn't say what I've run into are set-backs, but certainly I've encountered some stuff that needs investigated further.

What should be mentioned right now is that I'm working on a way to hopefully preserve downloads, so you don't waste a lot of time downloading, say, 200 mods, every time you want to change your list.

**Comment by Hypocrita20XX** [comment]
I am curious if I can merge the methods as well.
If there's a URL in the text field, use that, if not, use a link, add some check to ensure there's something in the file, etc.

**Comment by Hypocrita20XX** [comment]
Guess it's worth mentioning, my goal with 1.4.0 is to bring Conexus to a feature-complete state.
I'm counting on bugs being a thing afterwards, but nothing like these big stability changes/fixes and whatnot.

I might look into the translation feature after that, but I'm honestly more interesting in leaving development, ideally, in a few weeks' time.
Been going hard for a few months now and I'm very close to the point where I feel comfortable letting go of this project.

TLDR: Get this release out, get documentation polished, enter a feature freeze for a few weeks, fix bugs during that time, call it good, move on.

**Comment by Hypocrita20XX** [comment]
Had an epiphany, now Links.txt is much more flexible.
Instead of checking for *, it now just checks for steamcommunity.
Now * denotes mod names. Yes, I finally figured out a simple way to get mod names for those using the list of links method.

![image](https://user-images.githubusercontent.com/13149236/108914782-dffae380-75f9-11eb-92c6-3ff88f005d74.png)

Conexus can also handle if you don't include a mod name, as you can see in the picture, he'll default to the old format of order_id if no name is found. This does mean you need to have your mod names before your links, instead of after.
I mention that because that was necessary when I created link files from a collection, but I'll have to come up with a solution for that later.
Hopefully I explained that well, it'll all be in the updated documentation in clearer terms.

Practically this means formatting that file is much less restrictive, here's an example of a working file.
![image](https://user-images.githubusercontent.com/13149236/108914717-c78ac900-75f9-11eb-9537-319fd4259ca4.png)

**Comment by Hypocrita20XX** [comment]
Good news, all my tests are coming back positive. Hopefully will have a release tonight or tomorrow so long as the remaining tests come back fine.

**Comment by Hypocrita20XX** [comment]
Yeah ok, this release is ready for local, Github-only release.
